app/helpers/admin_helper.rb

```diff
@@ -22,26 +22,9 @@ def arrow_right
     icon("arrow-right")
   end
 
-  def request_both_links(info_request)
-    link_to(prominence_icon(info_request), request_path(info_request), title: 'View request on public website') + ' ' +
-      link_to(info_request.title, admin_request_path(info_request), title: 'View full details')
-  end
-
-  def public_body_both_links(public_body)
-    link_to(eye, public_body_path(public_body), :title => "view authority on public website") + " " +
-      link_to(h(public_body.name), admin_body_path(public_body), :title => "view full details")
-  end
-
-  def user_both_links(user)
-    link_to(prominence_icon(user.prominence), user_path(user), title: 'View user on public website') + ' ' +
-      link_to(h(user.name), admin_user_path(user), title: 'View full details')
-  end
-
-  def comment_both_links(comment)
-    link_to(prominence_icon(comment), comment_path(comment),
-            :title => "view comment on public website") + " " +
-      link_to(h(truncate(comment.body)), edit_admin_comment_path(comment),
-              :title => "view full details")
+  def both_links(record)
+    method = "#{record.class.to_s.underscore}_both_links"
+    send(method, record)
```

Comment:
Could check to see if method exists. if not raise argument error

Reply:
I was wondering about this, but eventually thought that the `NoMethodError` is kinda more descriptive (if you want to pass a record of class `X`, add the method named `x_both_links`) and it avoids a line of code. I'm happy that this breaks if you pass an "unexpected" record, as the only time someone will do that is while working on the codebase (as opposed to having to handle unknown user-submitted data).